ESPWebThingAdapter.h

```diff
@@ -360,22 +362,10 @@ class WebThingAdapter {
 
     JsonObject newAction = newBuffer->as<JsonObject>();
 
-    if (!newAction.containsKey(action->id)) {
-      b_has_body_data = false;
-      memset(body_data, 0, sizeof(body_data));
-      request->send(400);
-      delete newBuffer;
-      return;
-    }
-
-    ThingActionObject *obj = device->requestAction(newBuffer);
-
+    ThingActionObject *obj = action->create(newBuffer);
     if (obj == nullptr) {
-      b_has_body_data = false;
-      memset(body_data, 0, sizeof(body_data));
-      request->send(500);
-      delete newBuffer;
-      return;
+        request->send(404);
+        return;
```

Comment:
Can you expand why now are we returning a 404 where before this was a 500?

Reply:
You are right. That's wrong. I wanted to find the ThingActionObject first and returned a 404 when it wasn't able to find it. Was a bit confused. Good catch. Fixed it.

Reply:
Ok cool, don't we need to clear the memory as it was done previously? 

Also, previously we had an if statement just above this check, looking for the presence of the action id. In my understanding, this is removed due to the fact that it was something specific from the Withings protocol. Now we comply with the data payload described in the TD. right?

Reply:
As far as I understand the function is only called when the URL already exist. So, the 404 should return before that. Just checked it. As it turns out, this doesn't happen. But this check has to happen somewhere else. I will take a look into it. It should send a 404.

Reply:
So, it crashed before, because there was a generic handler for any /action/* endpoint, even if you didn't had a listener for it. I guess the Mozilla standard handled it different and returned an object with all available actions or something like that. I just removed it. You get the desired 404 now, if the action doesn't exist. Makes the code more simple as well.

Reply:
I added the clean memory part. That's a good point. Not necessary, but good to do. 
Yes, it now follows the TD payload structure and the webthings specific stuff is removed. Even though you can technically also define booleans etc. as input, but that cashes the code. Just added it as issue in #12